### PR TITLE
Medium editor

### PR DESCRIPTION
**Pull request by Nomafin**

### CODE REVIEW

File: src/Field/MediumEditor.php [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    protected $other_options = [];
+
+    /**
+     * Delay status
+     *
+     * @var integer
+     */
+    protected $delay = 0;
+
+    public function export( $register = false ) {
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: Add missing function comment.

File: src/Field/MediumEditor.php [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+     * @var string
+     */
+    protected $placeholder;
+
+    /**
+     * Standard buttons to show
+     *
+     * @var array
+     */
+    protected $standard_buttons = [];
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: What are the deafault buttons? Does the Medium Editor field define some defaults? Could the defaults be initialized here for code readability reasons?

File: src/Field/MediumEditor.php [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    public function add_button( $button ) {
+        $this->standard_buttons[] = $button;
+
+        $this->standard_buttons = array_unique( $this->standard_buttons );
+
+        return $this;
+    }
+
+    /**
+     * Remove a standard button
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: Maybe this comment could describe the visible end result of removing a button?

File: src/Field/MediumEditor.php [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+            throw new \Geniem\ACF\Exception( 'Geniem\ACF\Group: set_buttons() requires an array as an argument' );
+        }
+
+        $this->standard_buttons = $buttons;
+
+        return $this;
+    }
+
+    /**
+     * Add a standard button
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: Maybe there could be a description of the visible effect of adding a button?

File: src/Field/MediumEditor.php [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+     * Get placeholder value
+     *
+     * @return string
+     */
+    public function get_placeholder() {
+        return $this->placeholder;
+    }
+
+    /**
+     * Set buttons to show
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: What is the difference between this and the `add_button()` method? Maybe the comments could describe it?

File: src/Field/MediumEditor.php [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+     * Get set standard buttons
+     *
+     * @return array
+     */
+    public function get_buttons() {
+        return $this->standard_buttons;
+    }
+
+    /**
+     * Add a button to the custom buttons
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: What is the difference between a standard button and a custom button? Maybe describe it here?

File: src/Field/MediumEditor.php [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+     *
+     * @param  string $custom_button CustomButton name.
+     * @return \Geniem\ACF\Field\MediumEditor\CustomButton
+     */
+    public function get_custom_button( string $custom_button ) {
+        return $this->custom_buttons[ $custom_button ];
+    }
+
+    /**
+     * Get all custom buttons
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: What are custom buttons? Maybe describe them?

File: src/Field/MediumEditor.php [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+     * @var array
+     */
+    protected $custom_buttons = [];
+
+    /**
+     * Other Medium Editor options
+     *
+     * @var array
+     */
+    protected $other_options = [];
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: Maybe the default options could be initialized here?

File: src/Field/MediumEditor.php [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+     * Get set options
+     *
+     * @return array
+     */
+    public function get_options() {
+        return $this->other_options;
+    }
+
+    /**
+     * Get other choice values status
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: What is an `other choice`?

File: src/Field/MediumEditor.php [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+     *
+     * @return integer
+     */
+    public function get_other_choice() {
+        return $this->other_choice;
+    }
+
+    /**
+     * Delay loading until the field is clicked
```
Comment (villesiltala) [inline src/Field/MediumEditor.php]: Loading of what?

File: src/Field/MediumEditor/CustomButton.php [hunk 11]
```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Custom button class for ACF Medium Editor
```
Comment (villesiltala) [inline src/Field/MediumEditor/CustomButton.php]: What is a custom button? Maybe describe it here?

File: src/Field/MediumEditor/CustomButton.php [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+            throw new \Geniem\ACF\Exception( 'Geniem\ACF\Field\MediumEditor\CustomButton: set_attributes() requires an array as an argument' );
+        }
+
+        $this->attributes = $attributes;
+
+        return $this;
+    }
+
+    /**
+     * Add an attribute
```
Comment (villesiltala) [inline src/Field/MediumEditor/CustomButton.php]: What is an attribute?

File: src/Field/MediumEditor.php (hunk 2 above)
Comment (Nomafin) [reply src/Field/MediumEditor.php]: There are no defaults in the Medium Editor, every button must be activated manually.

File: src/Field/MediumEditor.php (hunk 8 above)
Comment (Nomafin) [reply src/Field/MediumEditor.php]: There are no default options here either.

File: src/Field/MediumEditor.php (hunk 2 above)
Comment (villesiltala) [reply src/Field/MediumEditor.php]: Okay 👍

File: src/Field/MediumEditor.php (hunk 8 above)
Comment (villesiltala) [reply src/Field/MediumEditor.php]: Ok 👍

**Pull request merged**